src/Exceptionless.Core/Services/StackService.cs

```diff
@@ -0,0 +1,113 @@
+                if (cancellationToken.IsCancellationRequested) break;
+
+                string organizationId = tuple.Item1, projectId = tuple.Item2, stackId = tuple.Item3;
+                string occurrenceCountCacheKey = GetStackOccurrenceCountCacheKey(organizationId, projectId, stackId),
+                    occurrenceMinDateCacheKey = GetStackOccurrenceMinDateCacheKey(organizationId, projectId, stackId),
+                    occurrenceMaxDateCacheKey = GetStackOccurrenceMaxDateCacheKey(organizationId, projectId, stackId);
+                var occurrenceCountTask = _cache.GetAsync<long>(occurrenceCountCacheKey, 0);
+                var occurrenceMinDateTask = _cache.GetAsync<double>(occurrenceMinDateCacheKey);
+                var occurrenceMaxDateTask = _cache.GetAsync<double>(occurrenceMaxDateCacheKey);
+
+                await Task.WhenAll(occurrenceCountTask, occurrenceMinDateTask, occurrenceMaxDateTask).AnyContext();
+                var occurrenceCount = (int)occurrenceCountTask.Result;
+                if (occurrenceCount == 0) continue;
+
+                await Task.WhenAll(
+                    _cache.DecrementAsync(occurrenceCountCacheKey, occurrenceCount, _expireTimeout),
+                    _cache.SetRemoveAsync(occurrenceSetCacheKey, tuple),
+                    _cache.RemoveAllAsync(new []{ occurrenceMinDateCacheKey, occurrenceMaxDateCacheKey })
+                ).AnyContext();
+
+                var occurrenceMinDate = occurrenceMinDateTask.Result.HasValue ? ConvertFromUnixTimeStamp(occurrenceMinDateTask.Result.Value) : SystemClock.UtcNow;
+                var occurrenceMaxDate = occurrenceMaxDateTask.Result.HasValue ? ConvertFromUnixTimeStamp(occurrenceMaxDateTask.Result.Value) : SystemClock.UtcNow;
+
+                try {
+                    if (!await _stackRepository.IncrementEventCounterAsync(organizationId, projectId, stackId, occurrenceMinDate, occurrenceMaxDate, occurrenceCount, sendNotifications).AnyContext()) {
+                        await IncrementStackUsage().AnyContext();
+                    }
+                    else {
+                        _logger.LogTrace("Increment event count {occurrenceCount} for organization:{organizationId} project:{projectId} stack:{stackId} with occurrenceMinDate:{occurrenceMinDate} occurrenceMaxDate:{occurrenceMaxDate}", occurrenceCount, organizationId, projectId, stackId, occurrenceMinDate, occurrenceMaxDate);
+                    }
+                }
+                catch {
+                    await IncrementStackUsage().AnyContext();
+                    throw;
+                }
+
+                Task IncrementStackUsage() {
+                    return IncrementStackUsageAsync(organizationId, projectId, stackId, occurrenceMinDate, occurrenceMaxDate, occurrenceCount);
+                }
+            }
+        }
+
+        internal double ConvertToUnixTimeStamp(DateTime dateUtc) {
+            return (dateUtc - _unixTimeStampOrigin).Ticks;
```

Comment:
Shouldn't we be able to just use the dateUtc ticks? instead of converting it every time? Or was there some kind of overflow?

Reply:
I tried this at the beginning, but the redis performs wired. when the number is to big, some bits will be lost. So I have to keep the number as small as possible.

Reply:
I tried to let `_unixTimeStampOrigin = new DateTime(1970,1,1)`, and the number retrieve from redis will be not accurate too. After let `_unixTimeStampOrigin = new DateTime(2000,1,1)`, all tests passed.